lib/class-wp-theme-json-resolver-gutenberg.php

```diff
@@ -571,7 +571,8 @@ public static function get_merged_data( $origin = 'custom' ) {
 			_deprecated_argument( __FUNCTION__, '5.9.0' );
 		}
 
-		$result = static::get_core_data();
+		$result = new WP_Theme_JSON();
```

Comment:
ugs, I just realized we should use the `WP_Theme_JSON_Gutenberg` class in Gutenberg. Prepared a fix at https://github.com/WordPress/gutenberg/pull/48648